java/google/registry/whois/WhoisCommandFactory.java

```diff
@@ -0,0 +1,53 @@
+// Copyright 2016 The Nomulus Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package google.registry.whois;
+
+import com.google.common.net.InternetDomainName;
+import google.registry.config.RegistryConfig.ConfigModule;
+
+import javax.annotation.Nullable;
+import java.net.InetAddress;
+
+/**
+ * A no-op base class used to provide custom whois commands.
+ * <p>To add custom commands, extend this class, then configure it in
+ * {@link ConfigModule#provideWhoisCommandFactoryClass()}.
+ */
+public class WhoisCommandFactory {
+
+  public WhoisCommand domainLookupCommand(InternetDomainName domainName) {
```

Comment:
Please use the same pattern that CustomLogicFactory uses for extensibility.  This is going about things quite differently, resulting in two different styles of extensibility frameworks that an implementer must learn.  In particular, the things that are missing are:

- A simple API in the factory, returning a single CustomLogic class per type of flow (or WHOIS command, in this case).
- ImmutableObject AutoValues for parameter/return value grouping.
- Noop extension points in base versions of the custom logic classes.
- Custom logic classes at all -- I don't see any equivalent to DomainCreateFlowCustomLogic et al here. I  would expect to see DomainLookupWhoisCustomLogic, NameserverLookupWhoisCustomLogic, etc.

It seems like you're thinking of overriding the WHOIS commands in their entirety, rather than plugging into a few points within them as with the flow custom logic.  That seems like a difference in approach that needs further exploration.  If you're just overriding the commands in their entirety, it's not clear to me why simply increasing visibility of some stuff like WhoisReader, and then replacing the /whois endpoint, wouldn't be simpler.